### PR TITLE
improve bigtable autoscaling sample

### PR DESCRIPTION
**Pull request by sergeylanzman**
* Fix bigtable.Client - add project
* Fix support multi instance in 1 project
* Add support multi cluster option
* Move min,max node , size_change_step to args

### CODE REVIEW

**Comment by andrewferlitsch** [comment]
@billyjacobson Following up -- do you have time to review this PR?

File: bigtable/metricscaler/metricscaler.py [hunk 1]
```diff
@@ -140,16 +143,16 @@ def main(
         'bigtable_instance',
         help='ID of the Cloud Bigtable instance to connect to.')
     parser.add_argument(
-        'bigtable_cluster',
-        help='ID of the Cloud Bigtable cluster to connect to.')
+        'bigtable_clusters',
+        help='ID of the Cloud Bigtable clusters to connect to.')
```
Comment (billyjacobson) [inline bigtable/metricscaler/metricscaler.py]: Comma separated list of IDs of Cloud Bigtable clusters to connect to.

File: bigtable/metricscaler/metricscaler.py [hunk 2]
```diff
@@ -160,13 +163,37 @@ def main(
...
+    # zone. If you need more quota, you can request more by following the
+    # instructions at https://cloud.google.com/bigtable/quota.
+    parser.add_argument(
+        '--max_node_count',
+        help='Maximum number of nodes',
+        default=30)
+    parser.add_argument(
+        '--size_change_step',
+        help='The number of nodes to change the cluster by',
+        default=5)
```
Comment (billyjacobson) [inline bigtable/metricscaler/metricscaler.py]: The default was 3 before, why did this change?

File: bigtable/metricscaler/metricscaler.py [hunk 3]
```diff
@@ -160,13 +163,37 @@ def main(
         help='How long to sleep in seconds between checking metrics after a '
              'scaling operation',
         default=60 * 10)
+    # The minimum number of nodes to use. The default minimum is 3. If you have
+    # a lot of data, the rule of thumb is to not go below 2.5 TB per node for
+    # SSD lusters, and 8 TB for HDD. The
```
Comment (billyjacobson) [inline bigtable/metricscaler/metricscaler.py]: s/lusters/clusters

File: bigtable/metricscaler/metricscaler.py [hunk 4]
```diff
@@ -140,16 +143,16 @@ def main(
...
     parser.add_argument(
-        'bigtable_cluster',
-        help='ID of the Cloud Bigtable cluster to connect to.')
+        'bigtable_clusters',
+        help='ID of the Cloud Bigtable clusters to connect to.')
     parser.add_argument(
         '--high_cpu_threshold',
         help='If Cloud Bigtable CPU usage is above this threshold, scale up',
-        default=0.6)
+        default=0.4)
```
Comment (billyjacobson) [inline bigtable/metricscaler/metricscaler.py]: Why are these defaults changing?

**Comment by andrewferlitsch** [comment]
@billyjacobson Thanks for reviewing the PR. I am still new to this -- is there a next step for me, or is it for the submitter of the PR?

File: bigtable/metricscaler/metricscaler.py (hunk 4 above)
Comment (sergeylanzman) [reply bigtable/metricscaler/metricscaler.py]: With replication mode CPU recommended 40%, but default can be 0.6

**Comment by sergeylanzman** [comment]
Thank you for code review, fixed all issues

File: bigtable/metricscaler/metricscaler.py [hunk 5]
```diff
@@ -26,7 +26,7 @@
 PROJECT = os.environ['GCLOUD_PROJECT']
 
 
-def get_cpu_load(cluster, instance):
+def get_cpu_load(instance, cluster):
     """Returns the most recent Cloud Bigtable CPU load measurement.
     Args:
             cluster (str): name of cluster
```
Comment (billyjacobson) [inline bigtable/metricscaler/metricscaler.py]: Swap the order of the Args comment too

File: bigtable/metricscaler/metricscaler.py [hunk 6]
```diff
@@ -118,7 +118,7 @@ def main(bigtable_instance,
     if_scaled = False
     for bigtable_cluster in bigtable_clusters:
         cluster_cpu = get_cpu_load(bigtable_cluster, bigtable_instance)
```
Comment (billyjacobson) [inline bigtable/metricscaler/metricscaler.py]: Swap the order of args here

**Comment by billyjacobson** [comment]
@crwilcox I'm adding you as a reviewer in case anything else needs to be done to get this merged. All of the changes look good to me, but I think the build might be failing and maybe you could help debug that if you have a chance. Thanks

File: bigtable/metricscaler/metricscaler.py (hunk 4 above)
Comment (billyjacobson) [reply bigtable/metricscaler/metricscaler.py]: oh, could you comment that somewhere? Or just link to the replication docs in a comment where they explain more about the parameters

Comment (sergeylanzman) [reply bigtable/metricscaler/metricscaler.py]: https://cloud.google.com/bigtable/docs/monitoring-instance

**Comment by andrewferlitsch** [comment]
@sergeylanzman Any update on the status of this merge?

**Comment by sergeylanzman** [comment]
Done!

**Comment by tmatsuo** [comment]
@sergeylanzman The build is failing. If you're still working on this, can you fix them? They're not very difficult to fix.

**Comment by tmatsuo** [comment]
Ok, I'll try to merge this tomorrow by directly push to `sergeylanzman:improve-bigtable-autoscaling`.

**Comment by tmatsuo** [comment]
@sergeylanzman 

I tried to push a fix, but I found out that it's little bit difficult without knowing more about this change.
Sad to say this, but we'll close this PR around the end of April unless we hear back from you.

**Comment by tmatsuo** [comment]
Closing

**Pull request closed without merging**